### PR TITLE
Change `TimestampInterval.minus` wrt "empty" intervals.

### PR DESCRIPTION
**Pull request by toddburnside**
There was some edge cases in `minus` related to intervals where `start === end`. 

In 2 cases, the resulting List of intervals would produce adjacent intervals that would "sum" to what I expect. Such as (using the test notation):
> List(interval(0, 3), interval(3, 10))

In 2 other cases, the result contained a "empty" interval disjoint from what I would expect. Such as:
> List(interval(0, 3), interval(10, 10))

I will put comments in the test denoting the exact behavior that changed.

@swalker2m - You'll have to let me know if this matches your expectations for your use in the `TimestampUnion` or not.

### CODE REVIEW

File: modules/tests/shared/src/test/scala/lucuma/core/util/TimestampIntervalSuite.scala [hunk 1]
```diff
@@ -125,16 +125,29 @@ class TimestampIntervalSuite extends DisciplineSuite {
     assertEquals(interval(1, 10).minus(interval(9, 11)), List(interval(1, 9)))
 
     // Proper Superset
-    assertEquals(interval(0, 10).minus(interval(3,  6)), List(interval(0, 3), interval(6, 10)))
+    assertEquals(interval(0, 10).minus(interval(3, 6)), List(interval(0, 3), interval(6, 10)))
+    assertEquals(interval(0, 10).minus(interval(0, 6)), List(interval(6, 10)))
```
Comment (toddburnside) [inline modules/tests/shared/src/test/scala/lucuma/core/util/TimestampIntervalSuite.scala:129]: This used to produce: `List(interval(0, 0), interval(6, 10))`

File: modules/tests/shared/src/test/scala/lucuma/core/util/TimestampIntervalSuite.scala [hunk 2]
```diff
@@ -125,16 +125,29 @@ class TimestampIntervalSuite extends DisciplineSuite {
...
+    assertEquals(interval(0, 10).minus(interval(3, 10)), List(interval(0, 3)))
```
Comment (toddburnside) [inline modules/tests/shared/src/test/scala/lucuma/core/util/TimestampIntervalSuite.scala:130]: This used to produce: List(interval(0, 3), interval(10, 10))

File: modules/tests/shared/src/test/scala/lucuma/core/util/TimestampIntervalSuite.scala [hunk 3]
```diff
@@ -125,16 +125,29 @@ class TimestampIntervalSuite extends DisciplineSuite {
...
+    // with empty intervals 
+    assertEquals(interval(0, 10).minus(interval(0, 0)), List(interval(0, 10)))
```
Comment (toddburnside) [inline modules/tests/shared/src/test/scala/lucuma/core/util/TimestampIntervalSuite.scala:132]: This used to produce: List(interval(0, 0), interval(0, 10))

File: modules/tests/shared/src/test/scala/lucuma/core/util/TimestampIntervalSuite.scala [hunk 4]
```diff
@@ -125,16 +125,29 @@ class TimestampIntervalSuite extends DisciplineSuite {
...
+    assertEquals(interval(0, 10).minus(interval(3, 3)), List(interval(0, 10)))
```
Comment (toddburnside) [inline modules/tests/shared/src/test/scala/lucuma/core/util/TimestampIntervalSuite.scala:133]: This used to produce: List(interval(0, 3), interval(3, 10))

File: modules/core/shared/src/main/scala/lucuma/core/util/TimestampInterval.scala [hunk 5]
```diff
@@ -85,12 +85,15 @@ sealed class TimestampInterval private (val start: Timestamp, val end: Timestamp
...
+      case Overlap.LowerPartial                                  => List(between(start, other.start))
+      case Overlap.UpperPartial                                  => List(between(other.end, end))
+      case Overlap.ProperSuperset if (other.start === other.end) => List(this)
+      case Overlap.ProperSuperset if (start === other.start)     => List(between(other.end, end))
+      case Overlap.ProperSuperset if (end === other.end)         => List(between(start, other.start))
+      case Overlap.ProperSuperset                                => List(between(start, other.start), between(other.end, end))
+      case Overlap.None                                          => List(this)
       case Overlap.ProperSubset |
-           Overlap.Equal          => Nil
+           Overlap.Equal                                         => Nil
```
Comment (swalker2m) [inline modules/core/shared/src/main/scala/lucuma/core/util/TimestampInterval.scala:96]: Hmm, I wonder if this should produce a `TimestampUnion` instead of a `List[TimestampInterval]`.  In the original code when the `List` is added to the union it gets merged in as expected.

Comment (toddburnside) [reply modules/core/shared/src/main/scala/lucuma/core/util/TimestampInterval.scala:96]: So the disjoint "empty" intervals didn't cause a problem? It seems like that would create a Union with 2 pieces. But, maybe the empty piece is expected?

Comment (toddburnside) [reply modules/core/shared/src/main/scala/lucuma/core/util/TimestampInterval.scala:96]: Expected by your code, that is. Mine doesn't expect it. But, I could filter it out.

Comment (swalker2m) [reply modules/core/shared/src/main/scala/lucuma/core/util/TimestampInterval.scala:96]: An empty interval would add nothing to the sum, so in that sense it doesn't matter for time accounting.

**Review by swalker2m — APPROVED** [review]
Thanks, this seems fine. The number of cases is annoying but your result is much more intuitive. :+1:

**Comment by toddburnside** [comment]
> Thanks, this seems fine. The number of cases is annoying but your result is much more intuitive. 👍

OK. Fortunately, it only seems to be odd wrt `minus`.
